### PR TITLE
[IOTDB-1552]Some basic operators should not be applied to text time series

### PR DESCRIPTION
**Pull request by ijihang**
JIRA:https://issues.apache.org/jira/browse/IOTDB-1552

For text type,Only equal and not equal should be supported

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/BasicFunctionOperator.java [hunk 1]
```diff
@@ -100,13 +100,18 @@ public void reverseFunc() {
...
         break;
       case TEXT:
-        ret =
-            funcToken.getUnaryExpression(
-                singlePath,
-                (value.startsWith("'") && value.endsWith("'"))
-                        || (value.startsWith("\"") && value.endsWith("\""))
-                    ? new Binary(value.substring(1, value.length() - 1))
-                    : new Binary(value));
+        if (funcToken.name().equals("EQ") || funcToken.name().equals("NOTEQUAL")) {
```
Comment (wangchao316) [inline server/src/main/java/org/apache/iotdb/db/qp/logical/crud/BasicFunctionOperator.java:103]: BasicOperatorType.EQ   replaces “EQ”       
 BasicOperatorType.NOTEQUAL  replaces   "NOTEQUAL"

Comment (ijihang) [reply server/src/main/java/org/apache/iotdb/db/qp/logical/crud/BasicFunctionOperator.java]: Thanks,new commit modify

Comment (ijihang) [reply server/src/main/java/org/apache/iotdb/db/qp/logical/crud/BasicFunctionOperator.java]: Thanks, new commit modify

File: server/src/test/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillDataSetTest.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
   private String[] sqls = {
     "SET STORAGE GROUP TO root.vehicle",
     "CREATE TIMESERIES root.vehicle.d0.s0 WITH DATATYPE=INT32, ENCODING=RLE",
-    "CREATE TIMESERIES root.vehicle.d0.s1 WITH DATATYPE=TEXT, ENCODING=PLAIN",
+    "CREATE TIMESERIES root.vehicle.d0.s1 WITH DATATYPE=INT32, ENCODING=RLE",
```
Comment (qiaojialin) [inline server/src/test/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillDataSetTest.java:45]: Hi, why change this?

Comment (ijihang) [reply server/src/test/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillDataSetTest.java:45]: SQL:   "select last_value(s0) from root.vehicle.* where s1 > 1 or s0 > 1  group by ([0,20), 1ms) fill (int32[Previous]) order by time desc"    "select last_value(s0) from root.vehicle.* where s1 > 1  group by ([0,20), 1ms) fill (int32[Previous]) order by time desc"    s1 is TEXT

**Review by qiaojialin — APPROVED** [review]

**Review by wangchao316 — APPROVED** [review]
LGTM

**Pull request merged**